public/js/angular/controllers.js

```diff
@@ -276,6 +276,96 @@ angular
         });
     }
   ])
+  .controller('createUpdateBadgeController', ['$rootScope', '$scope', '$http', '$routeParams', '$location', 'wmNav', 'CONFIG',
+    function ($rootScope, $scope, $http, $routeParams, $location, wmNav, config) {
+      wmNav.page('create-badge');
+      wmNav.section('explore');
+
+      // Update or create?
+      $scope.view = $routeParams.badge ? 'update' : 'create';
+      $scope.title = $routeParams.badge ? 'Update Badge' : 'Create a New Badge';
+
+      if ($scope.view === 'update') {
+        $http
+          .get('/api/badges/' + $routeParams.badge)
+          .success(function (data) {
+            data.tags = data.tags.map(function (obj) {
+              return obj.value;
+            }).join(', ');
+            $scope.badge = data;
+          })
+          .error(function (err) {
+            console.log(err);
+          });
+      } else {
+        // This holds all the values for the new badge
+        $scope.badge = {
+          criteria: [{
+            description: '# This is an example\n* point 1\n* point 2\n* point 3'
+          }]
+        };
+      }
+
+      $scope.criteriaPreview = function () {
+        if (!$scope.badge || !$scope.badge.criteria) {
+          return '';
+        }
+        var result = $scope.badge.criteria.map(function (item) {
+          return item.description;
+        }).join('\n\n');
+        return result;
+      };
+
+      $scope.addCriterion = function () {
+        $scope.badge.criteria.push({
+          description: ''
+        });
+      };
+
+      $scope.removeCriterion = function (index) {
+        $scope.badge.criteria.splice(index, 1);
+      };
+
+      function prepareBadge(data) {
+        var badge = angular.copy(data);
+
+        // Badgekit api requires both, but we want them to be the same.
+        badge.consumerDescription = data.earnerDescription;
+        badge.criteriaUrl = 'https://webmaker.org/badges/' + badge.slug;
+        badge.unique = 0;
+        badge.type = 'Skill';
+
+        // Tags
+        badge.tags = data.tags.split(/[\s,]+/).filter(function (item) {
+          return item;
+        });
+
+        // Criteria
+        badge.criteria = data.criteria.map(function (criterion) {
+          criterion.required = 1;
+          return criterion;
+        });
+
+        return badge;
+      }
+
+      $scope.submit = function (badge) {
+        var url = '/api/badges/' + ($routeParams.badge ? $routeParams.badge + '/' + $scope.view : 'create');
+        badge = prepareBadge(badge);
+        console.log('Sending...', badge);
+        $http
+          .post(url, badge)
+          .success(function (data) {
+            console.log('Success!', data);
```

Comment:
i would probably keep it for debugging purposes since this is an admin page anyway